contracts/contracts/FeeSub.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/token/ERC20/IERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/token/ERC20/utils/SafeERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/access/Ownable.sol";
+
+import "./RequestManager.sol";
+
+contract FeeSub is Ownable {
+    using SafeERC20 for IERC20;
+
+    RequestManager public requestManager;
+    mapping(address token => uint256) public minimumAmounts;
+    mapping(bytes32 requestId => address sender) public senders;
+
+    constructor(address _requestManager) {
+        requestManager = RequestManager(_requestManager);
+    }
+
+    function createRequest(
+        uint256 targetChainId,
+        address sourceTokenAddress,
+        address targetTokenAddress,
+        address targetAddress,
+        uint256 amount,
+        uint256 validityPeriod
+    ) external returns (bytes32) {
+        require(
+            minimumAmounts[sourceTokenAddress] > 0,
+            "Token not to be subsidized"
+        );
+        require(
+            amount >= minimumAmounts[sourceTokenAddress],
+            "Transfer amount too small to be subsidized"
+        );
+
+        require(
+            IERC20(sourceTokenAddress).allowance(msg.sender, address(this)) >=
+                amount,
+            "Insufficient allowance"
+        );
+
+        IERC20(sourceTokenAddress).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount
+        );
+
+        bytes32 requestId = requestManager.createRequest(
+            targetChainId,
+            sourceTokenAddress,
+            targetTokenAddress,
+            targetAddress,
+            amount,
+            validityPeriod
+        );
+        senders[requestId] = msg.sender;
+
+        return requestId;
+    }
+
+    function transferableAmount(
```

Comment:
We figured out that we don't need this function

Reply:
I am not sure about this one. Can you remind me why we decided to not use this fn ?

Reply:
@fredo

Reply:
already explained him in DM. 
We do not need this function cause it's trivial. If we were to use this contract (which is if transfer amount >= minimumAmount) then the transferable amount = transfer amount. 
If the condition is false we are not using the contract anyway.